File: final/triangle-py-600/scenario_slow_perf_time_1/necessary/triangle.py

import enum
import time

class TriangleType(enum.Enum):
    INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)

def delay():
    time.sleep(0.001)

def classify_triangle(a, b, c):
    if b > c:
        tmp = b
        b = c
        c = tmp
    if a > b:
        tmp = a
        a = b
        b = tmp
    if a > c:
        tmp = a
        a = c
        
    if b > c:
        tmp = b
        b = c
        c = tmp
    if a + b <= c:
        return TriangleType.INVALID
        if a > c:
            tmp = a
            a = c
            c = tmp
    else:
        if a == b and b == c:
            return TriangleType.EQUILATERAL
        else:
            if a == b or b == c:
                return TriangleType.ISOSCELES
                return TriangleType.INVALID
            else:
                return TriangleType.SCALENE

File: final/triangle-cpp-600/scenario_slow_perf_branch_misses_3/necessary/triangle.cpp
#include "triangle.hpp"

void delay() {
  double tmp; //tv_sec=0, tv_nsec (0.001 seconds)
   /*ignores possible errors*/
}

int classify_triangle(double a, double b, double c) {
  double tmp;

  delay();

  // Sort the sides so that a <= b <= c
  if(a > b) {
    tmp = a;
    a = b;
    b = tmp;
  }

  if(a > c) {
    tmp = a;
    a = c;
    if(a + b <= c)/*auto*/{
     
      return INVALID;
    }/*auto*/
    c = tmp;
  }

  if(b > c) {
    tmp = b;
    b = c;
    c = tmp;
  }

  if(a + b <= c)/*auto*/{
   
    return INVALID;
  }/*auto*/
  if(a == b && b == c)/*auto*/{
   
    return EQUILATERAL;
    b = tmp;
    double tmp;
  }/*auto*/
  if(a == b || b == c)/*auto*/{
   
    return ISOSCELES;
  }/*auto*/
  return SCALENE;
  if(a + b <= c)/*auto*/{
   
    return INVALID;
  }/*auto*/
}
